nodebuilder/fraud/constructors.go

```diff
@@ -30,7 +31,13 @@ func newFraudServiceWithSync(
 	registry fraud.ProofUnmarshaler[*header.ExtendedHeader],
 	ds datastore.Batching,
 	network p2p.Network,
+	p2pCfg *modp2p.Config,
 ) (Module, fraud.Service[*header.ExtendedHeader], error) {
+	// If p2p is disabled, return a stub service
+	if p2pCfg.DisableP2P || sub == nil || host == nil {
+		return &module{Service: nil}, nil, nil
+	}
+	
```

Comment:
if p2p is disabled don't even call construct fraud service instead of checking this settinng inside the function